File: ONLINE_CODING_TEST/school.programmers/lvl_2/12981_영어_끝말잇기.java
import java.util.ArrayList;
import java.util.List;

class Solution {
  public int[] solution(int n, String[] words) {
    int[] answer = new int[2];
    List<String> list = new ArrayList<String>();
    String curr, prev;
    
    for (int i = 0; i < words.length; i++) {
      curr = words[i];
      
      if (i > 0) {
        prev = words[i - 1];
        // 이전에 등장했는가?
        boolean isDupl = list.contains(curr);
        // 끝말 잇기가 맞는가?
        boolean isWordChain = prev.charAt(prev.length() - 1) == curr.charAt(0);
        
        // 중복이거나, 끝말 잇기가 성립 안된다면
        if (isDupl || !isWordChain) {
          // 회차
          answer[0] = (i % n) + 1;
          // 차례
          answer[1] = (i / n) + 1;
          break;
        }
      }
      
      list.add(curr);
    }

    return answer;
  }
}

File: ONLINE_CODING_TEST/school.programmers/lvl_2/12981_영어_끝말잇기.js
function solution(n, words) {
  let answer = [0, 0];
  let list = [];
  let curr, prev;

  for (let i = 0; i < words.length; i++) {
    curr = words[i];

    if (i > 0) {
      prev = words[i - 1];

      // 이전에 등장 했는가?
      let isDupl = list.includes(curr);
      // 끝믈 잇기가 맞는가?
      let isWordChain = prev[prev.length - 1] == curr[0];

      if (isDupl || !isWordChain) {
        answer[0] = (i % n) + 1;
        answer[1] = Math.floor(i / n) + 1;
        break;
      }
    }

    list.push(curr)
  }

  return answer;
}

const tests = [{
    n: 3,
    words: ["tank", "kick", "know", "wheel", "land", "dream", "mother", "robot", "tank"]
  },
  {
    n: 5,
    words: ["hello", "observe", "effect", "take", "either", "recognize", "encourage", "ensure", "establish", "hang", "gather", "refer", "reference", "estimate", "executive"]
  },
  {
    n: 2,
    words: ["hello", "one", "even", "never", "now", "world", "draw"]
  },
];

for (const test of tests) {
  console.log('-'.repeat(100));
  console.log('param : ', test, '\n');
  console.time();
  console.log('\n\t->\t', solution(test.n, test.words));
  console.timeEnd();
  console.timeLog();
  console.log('-'.repeat(100) + '\n');
}